### PR TITLE
Dtspo 11343 nameoverrides

### PR DESCRIPTION
**Pull request by darren221980**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DTSPO-11347

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: apps/admin/traefik-no-proxy/kustomization.yaml [hunk 1]
```diff
@@ -1,4 +1,6 @@
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
+nameSuffix: -no-proxy
 resources:
+  - ../traefik2/traefik2.yaml
   - traefik-no-proxy.yaml
```
Comment (reespozzi) [inline apps/admin/traefik-no-proxy/kustomization.yaml:6]: Can this be moved under `patchesStrategicMerge`? Causes two separate helm releases currently

File: apps/admin/traefik-auth-proxy/kustomization.yaml [hunk 2]
```diff
@@ -1,8 +1,6 @@
 apiVersion: kustomize.config.k8s.io/v1beta1
 kind: Kustomization
+nameSuffix: -auth-proxy
 resources:
+  - ../traefik2/traefik2.yaml
   - traefik-auth-proxy.yaml
```
Comment (reespozzi) [inline apps/admin/traefik-auth-proxy/kustomization.yaml:6]: Same as the no-proxy patch comment here too

**Review by reespozzi — COMMENTED** [review]
Could we also look at removing https://github.com/hmcts/sds-flux-config/blob/c12efd0afcea789d924909682d18dafd0afc44de/apps/admin/traefik-auth-proxy/traefik-auth-proxy.yaml#L20-L59 from the auth-proxy and no-proxy patch files as a lot of the code is duplicate of the base `traefik2.yaml` file?

File: apps/admin/traefik-no-proxy/traefik-no-proxy.yaml [hunk 3]
```diff
@@ -7,15 +7,6 @@ metadata:
...
-      chart: traefik
-      # update the crd version in traefik-crds when updating this
-      version: 10.6.0
-      sourceRef:
-        kind: HelmRepository
-        name: traefik
-        namespace: admin
   interval: 1m
   values:
     additionalArguments:
```
Comment (reespozzi) [inline apps/admin/traefik-no-proxy/traefik-no-proxy.yaml:12]: Also likely to need 
```
   - --entryPoints.web.forwardedHeaders.insecure
   - --entryPoints.websecure.forwardedHeaders.insecure
```
In this file too

File: apps/admin/traefik2-auth-middlewares/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
```
Comment (reespozzi) [inline apps/admin/traefik2-auth-middlewares/kustomization.yaml:2]: Has this been added to the environment kustomization anywhere? similar to https://github.com/hmcts/cnp-flux-config/blob/master/apps/admin/demo/base/kustomization.yaml#L18

**Review by reespozzi — APPROVED** [review]
Looks good, checked the manifests before and after and they match up for traefik instances and middlewares etc, only question is do we need the rename to traefik2-no-proxy/auth-proxy instead of traefik-no-proxy/auth-proxy as it was before?

**Comment by reespozzi** [comment]
> Looks good, checked the manifests before and after and they match up for traefik instances and middlewares etc, only question is do we need the rename to traefik2-no-proxy/auth-proxy instead of traefik-no-proxy/auth-proxy as it was before?

-
release is named as traefik2 in https://github.com/hmcts/sds-flux-config/blob/DTSPO-11343-nameoverrides/apps/admin/traefik2/traefik2.yaml otherwise if left as traefik it errors out as not being able to find that release.

**Review by adusumillipraveen — APPROVED** [review]

**Pull request merged**